src/Streamly/Internal/Data/Stream/StreamD/Nesting.hs

```diff
@@ -4011,4 +4013,449 @@ joinLeftMerge cmp (Stream stepa ta) (Stream stepb tb) =
 
     step _ (_, _, _, _, _, _, _, _, _) = do
         -- liftIO $ print "Step 11"
-        return Stop
\ No newline at end of file
+        return Stop
+
+-------------------------------------------------------------------------------
+-- Outer Join sorted streams --------------------------------------------------
+-------------------------------------------------------------------------------
+
+{-# INLINE_NORMAL joinOuterMerge #-}
+joinOuterMerge :: (MonadIO m, Eq a, Eq b) =>
+    (a -> b -> Ordering)
+    -> Stream m a
+    -> Stream m b
+    -> Stream m (Maybe a, Maybe b)
+joinOuterMerge cmp (Stream stepa ta) (Stream stepb tb) =
+    Stream
+    step
+    (Just ta, Just tb, Nothing, Nothing, Nothing, Nothing, Nothing, LeftRun, 0)
```

Comment:
A custom data type would be more cleaner I believe

Reply:
Would be easier to understand and review

Reply:
Can you document this state a little bit. Can you describe what each positional argument is? It would be easier tor review that way.

Reply:
Documented the state. Tried to create a custom data type but had an issue with type inference due to the "pattern synonym" of Stream.